pkg/controller/interceptor/interceptors.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package interceptor offers reconciler interceptors
+package interceptor
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+)
+
+// LoggingInterceptor wraps a Reconciler and logs messages
+// before and after the reconciliation,
+// reporting if the operation was successful together with its duration.
+func LoggingInterceptor(log logging.Logger, r reconcile.Reconciler) reconcile.Func {
+	return func(ctx context.Context, req reconcile.Request) (res reconcile.Result, err error) {
+		l := log.WithValues("request", req)
+		l.Info("Reconciling")
```

Comment:
I don't think we should include this log line. It should be enough to log only when the reconcile completes. That log line contains all the same information, with the added benefit of knowing the duration.

The only reasons I can think of to _also_ log when a reconcile starts would be:

* You're particularly concerned about emitting a log line before a potential crash in the reconciler.
* You're actively watching logs and want to see when something is happening.

For the above cases debug logging should be enough, and I'm 99% sure controller-runtime already emits a debug log when each reconcile starts.

Remember that logs are expensive to emit, store, etc relative to other metrics. This is part of why I believe we should be very sparing with what we emit at info level.